text/0000-api-version-compat.md

```diff
@@ -0,0 +1,119 @@
+Because platform authors may want to shield end users from deprecation warning caused by the platform's use of deprecated
+lifecycle features, the lifecycle will provide the following options to configure deprecation warnings:
+
+All lifecycle binaries will have a deprecation mode, configurable in the following way:
+| Flag              | Environment Variable | Default
+|-------------------|----------------------|--------
+| -deprecation-mode | CNB_DEPRECATION_MODE | all
+
+The allowed values of deprecation mode are as follows:
+| Mode      | Behavior
+|-----------|---------
+| all       | prints all deprecation warnings
+| buildpack | prints only buildpack API deprecation warnings
+| platform  | prints only platform API deprecation warnings
+| none      | silences all deprecation warnings
+
+# How it Works
+[how-it-works]: #how-it-works
+## Spec
+The core team will ensure that no changes are added to a minor spec release that imply incompatibility with the previous minor.
+
+Examples of ALLOWED changes in a minor 0.x API version:
+1. the addition of new image labels
+1. the addition of a key to an existing label
+1. the addition of a key to an existing file
+1. rename a file or a key in a file supplied as input
+1. rename a flag
+1. remove a flag
+
+Example of DISALLOWED changes in a minor API version:
+1. changes to the contractual build plan
+2. type changes to an existing key in a label or output file
+
+## Lifecycle
+The lifecycle will print deprecation warnings according to the rules described above. The lifecycle will
+guarantee that all usage that conforms to API `0.n` will continue to work, even when indicating API `0.n+1` is implemented.
+
+## Platforms
+Platforms that wish to enforce compatibility should now treat all minor API versions as non-breaking with the exception
+of buildpack and platform `0.1` -> `0.2`. These API versions did include breaking changes and will not be changed.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+The spec and lifecycle now carry the burden of ensuring there are no breaking changes before `1.0`. `1.0` is an important symbolic
+milestone and there could be unforseen and compelling reasons to make a breaking change before we are ready to release `1.0` APIs.
+
+# Alternatives
+[alternatives]: #alternatives
+
+- Keep doing what we are doing
+    - This will continue to cause pain for platform and buildpack authors
+- Continue to assume that minor API versions are breaking, but allow the lifecycle to indicate support for multiple versions simultaneously and use this to ease the disruption caused by new APIs 
+    - We will likely want to support multiple APIs in the lifecycle eventually to ease the transition to 1.0
+
+# Prior Art
+[prior-art]: #prior-art
+
+
+# Unresolved Questions
+[unresolved-questions]: #unresolved-questions
+
+- Will we allow removal of features in a minor API version in the spec itself post `1.0`?
```

Comment:
I don't understand this, can you elaborate?

Reply:
I believe it's asking, "If we are going to follow this same versioning scheme and semantics for 1.0+?"

Reply:
Can the spec remove a feature from an API between the 1.0 and 1.1 versions. This RFC says feature removal is allowed between 0.4 and 0.5 for example, but makes no comment about spec changes post 1.0.